yesod-core/Yesod/Core/Handler.hs

```diff
@@ -1360,6 +1361,16 @@ rawRequestBody = do
 fileSource :: MonadResource m => FileInfo -> ConduitT () S.ByteString m ()
 fileSource = transPipe liftResourceT . fileSourceRaw
 
+-- | Strict `ByteString` body from `FileInfo`.
+-- This function blocking while read file.
+--
+-- > do
+-- >     fileByteString <- fileSourceByteString fileInfo
+--
+-- @since 1.6.4
+fileSourceByteString :: MonadResource m => FileInfo -> m S.ByteString
+fileSourceByteString fileInfo = fileSource fileInfo `connect` CL.foldMap id
```

Comment:
There are a few problems with this implementation:

1. `CL.foldMap id` has quadratic complexity. It would be better to use something like [`sinkLazy`](https://www.stackage.org/haddock/lts-11.5/conduit-1.3.0.2/Conduit.html#v:sinkLazy)
2. It's more common to use `runConduit $ src .| sink`

How about:

```haskell
runConduit $ fileSource fileInfo .| Conduit.

Reply:
I use `runConduit`.
But `fileSource` function result strict `ByteString`.
So I think can't use `sinkLazy`.
Do you think?

Reply:
You'd have to use `Data.ByteString.toStrict`, but that's more efficient than using `foldMap`.